Intermitent test failure _create_shared_array RawArray test_executed_xy_par_1024_1024

### Summary

I have seen this a couple of times on github actions, e.g.
https://github.com/mantidproject/mantidimaging/pull/973/checks?check_run_id=2564558078

### Steps To Reproduce

Unknown

### Expected Behaviour

Pass

### Current Behaviour

```
FAILED mantidimaging/core/operations/rebin/test/rebin_test.py::RebinTest::test_executed_xy_par_1024_1024

Fatal Python error: Bus error
```

### Context

Happens occasionally on PR tests. I've seen it as far back as April 12.

### Failure Logs

```
2021-05-12T08:48:13.3789470Z + CMD='xvfb-run pytest -n auto --count 10 --ignore=mantidimaging/eyes_tests'
2021-05-12T08:48:13.3790572Z + ARGS=
2021-05-12T08:48:13.3791321Z + runuser '--command=xvfb-run pytest -n auto --count 10 --ignore=mantidimaging/eyes_tests '
2021-05-12T08:48:19.9108256Z ============================= test session starts ==============================
2021-05-12T08:48:19.9109715Z platform linux -- Python 3.8.8, pytest-6.2.1, py-1.10.0, pluggy-0.13.1
2021-05-12T08:48:19.9110427Z Using --randomly-seed=1908370704
2021-05-12T08:48:19.9110904Z rootdir: /opt/mantidimaging
2021-05-12T08:48:19.9111690Z plugins: cov-2.10.1, randomly-3.5.0, xdist-2.2.0, repeat-0.9.1, forked-1.3.0
2021-05-12T08:48:19.9112386Z gw0 I / gw1 I
2021-05-12T08:48:19.9112707Z gw0 [1231] / gw1 [1231]
2021-05-12T08:48:19.9112932Z 
2021-05-12T08:48:20.2086660Z ......................................................................... [  5%]
2021-05-12T08:48:21.2262256Z ......................................................................... [ 11%]
2021-05-12T08:48:23.2563852Z ........................................................................ [ 17%]
2021-05-12T08:48:23.7619200Z ........................................................................ [ 23%]
2021-05-12T08:48:24.1062721Z ........................................................................ [ 29%]
2021-05-12T08:48:24.3629326Z ........................................................................ [ 35%]
2021-05-12T08:48:24.6149273Z ........................................................................ [ 41%]
2021-05-12T08:48:25.0947381Z ........................................................................ [ 46%]
2021-05-12T08:48:26.3374110Z ................................................sssssss................. [ 52%]
2021-05-12T08:48:26.7023780Z ........................................................................ [ 58%]
2021-05-12T08:48:26.9843960Z ......................................................................... [ 64%]
2021-05-12T08:48:27.1999121Z ........................................................................ [ 70%]
2021-05-12T08:48:27.4794890Z ........................................................................ [ 76%]
2021-05-12T08:48:27.9411785Z ........................................................................ [ 82%]
2021-05-12T08:48:30.9435105Z ........................................ss.............................. [ 87%]
2021-05-12T08:48:34.1424024Z ........................................................................ [ 93%]
2021-05-12T08:48:35.0285112Z ..............s.......................Fatal Python error: Bus error
2021-05-12T08:48:35.0286424Z 
2021-05-12T08:48:35.0287169Z Thread 0x00007f2e2ab92700 (most recent call first):
2021-05-12T08:48:35.0331046Z   File "/opt/miniconda/envs/mantidimaging_test/lib/python3.8/site-packages/execnet/gateway_base.py", line 400 in read
2021-05-12T08:48:35.0364057Z   File "/opt/miniconda/envs/mantidimaging_test/lib/python3.8/site-packages/execnet/gateway_base.py", line 432 in from_io
2021-05-12T08:48:35.0365678Z   File "/opt/miniconda/envs/mantidimaging_test/lib/python3.8/site-packages/execnet/gateway_base.py", line 967 in _thread_receiver
2021-05-12T08:48:35.0367322Z   File "/opt/miniconda/envs/mantidimaging_test/lib/python3.8/site-packages/execnet/gateway_base.py", line 220 in run
2021-05-12T08:48:35.0415775Z   File "/opt/miniconda/envs/mantidimaging_test/lib/python3.8/site-packages/execnet/gateway_base.py", line 285 in _perform_spawn
2021-05-12T08:48:35.0419747Z 
2021-05-12T08:48:35.0420435Z Current thread 0x00007f2e2c104740 (most recent call first):
2021-05-12T08:48:35.0424517Z   File "/opt/miniconda/envs/mantidimaging_test/lib/python3.8/multiprocessing/sharedctypes.py", line 62 in RawArray
2021-05-12T08:48:35.0426238Z   File "/opt/miniconda/envs/mantidimaging_test/lib/python3.8/multiprocessing/sharedctypes.py", line 88 in Array
2021-05-12T08:48:35.0427451Z   File "/opt/miniconda/envs/mantidimaging_test/lib/python3.8/multiprocessing/context.py", line 141 in Array
2021-05-12T08:48:35.0433966Z   File "/opt/mantidimaging/mantidimaging/core/parallel/utility.py", line 77 in _create_shared_array
2021-05-12T08:48:35.0435181Z   File "/opt/mantidimaging/mantidimaging/core/parallel/utility.py", line 45 in create_array
2021-05-12T08:48:35.0436296Z   File "/opt/mantidimaging/mantidimaging/core/operations/rebin/rebin.py", line 167 in _create_reshaped_array
2021-05-12T08:48:35.0437432Z   File "/opt/mantidimaging/mantidimaging/core/operations/rebin/rebin.py", line 53 in filter_func
2021-05-12T08:48:35.0438885Z   File "/opt/mantidimaging/mantidimaging/core/operations/rebin/test/rebin_test.py", line 101 in do_execute_xy
2021-05-12T08:48:35.0440056Z   File "/opt/mantidimaging/mantidimaging/core/operations/rebin/test/rebin_test.py", line 80 in test_executed_xy_par_1024_1024
2021-05-12T08:48:35.0441212Z   File "/opt/miniconda/envs/mantidimaging_test/lib/python3.8/unittest/case.py", line 633 in _callTestMethod
2021-05-12T08:48:35.0443143Z   File "/opt/miniconda/envs/mantidimaging_test/lib/python3.8/unittest/case.py", line 676 in run
2021-05-12T08:48:35.0457168Z   File "/opt/miniconda/envs/mantidimaging_test/lib/python3.8/unittest/case.py", line 736 in __call__
2021-05-12T08:48:35.0458845Z   File "/opt/miniconda/envs/mantidimaging_test/lib/python3.8/site-packages/_pytest/unittest.py", line 321 in runtest
2021-05-12T08:48:35.0460484Z   File "/opt/miniconda/envs/mantidimaging_test/lib/python3.8/site-packages/_pytest/runner.py", line 162 in pytest_runtest_call
2021-05-12T08:48:35.0461773Z   File "/opt/miniconda/envs/mantidimaging_test/lib/python3.8/site-packages/pluggy/callers.py", line 187 in _multicall
2021-05-12T08:48:35.0463006Z   File "/opt/miniconda/envs/mantidimaging_test/lib/python3.8/site-packages/pluggy/manager.py", line 84 in <lambda>
2021-05-12T08:48:35.0464226Z   File "/opt/miniconda/envs/mantidimaging_test/lib/python3.8/site-packages/pluggy/manager.py", line 93 in _hookexec
2021-05-12T08:48:35.0465692Z   File "/opt/miniconda/envs/mantidimaging_test/lib/python3.8/site-packages/pluggy/hooks.py", line 286 in __call__
2021-05-12T08:48:35.0466910Z   File "/opt/miniconda/envs/mantidimaging_test/lib/python3.8/site-packages/_pytest/runner.py", line 255 in <lambda>
2021-05-12T08:48:35.0468719Z   File "/opt/miniconda/envs/mantidimaging_test/lib/python3.8/site-packages/_pytest/runner.py", line 311 in from_call
2021-05-12T08:48:35.0477333Z   File "/opt/miniconda/envs/mantidimaging_test/lib/python3.8/site-packages/_pytest/runner.py", line 254 in call_runtest_hook
2021-05-12T08:48:35.0478971Z   File "/opt/miniconda/envs/mantidimaging_test/lib/python3.8/site-packages/_pytest/runner.py", line 215 in call_and_report
2021-05-12T08:48:35.0480311Z   File "/opt/miniconda/envs/mantidimaging_test/lib/python3.8/site-packages/_pytest/runner.py", line 126 in runtestprotocol
2021-05-12T08:48:35.0489675Z   File "/opt/miniconda/envs/mantidimaging_test/lib/python3.8/site-packages/_pytest/runner.py", line 109 in pytest_runtest_protocol
2021-05-12T08:48:35.0490981Z   File "/opt/miniconda/envs/mantidimaging_test/lib/python3.8/site-packages/pluggy/callers.py", line 187 in _multicall
2021-05-12T08:48:35.0492205Z   File "/opt/miniconda/envs/mantidimaging_test/lib/python3.8/site-packages/pluggy/manager.py", line 84 in <lambda>
2021-05-12T08:48:35.0493892Z   File "/opt/miniconda/envs/mantidimaging_test/lib/python3.8/site-packages/pluggy/manager.py", line 93 in _hookexec
2021-05-12T08:48:35.0495133Z   File "/opt/miniconda/envs/mantidimaging_test/lib/python3.8/site-packages/pluggy/hooks.py", line 286 in __call__
2021-05-12T08:48:35.0496313Z   File "/opt/miniconda/envs/mantidimaging_test/lib/python3.8/site-packages/xdist/remote.py", line 89 in run_one_test
2021-05-12T08:48:35.0497761Z   File "/opt/miniconda/envs/mantidimaging_test/lib/python3.8/site-packages/xdist/remote.py", line 72 in pytest_runtestloop
2021-05-12T08:48:35.0499097Z   File "/opt/miniconda/envs/mantidimaging_test/lib/python3.8/site-packages/pluggy/callers.py", line 187 in _multicall
2021-05-12T08:48:35.0500304Z   File "/opt/miniconda/envs/mantidimaging_test/lib/python3.8/site-packages/pluggy/manager.py", line 84 in <lambda>
2021-05-12T08:48:35.0501530Z   File "/opt/miniconda/envs/mantidimaging_test/lib/python3.8/site-packages/pluggy/manager.py", line 93 in _hookexec
2021-05-12T08:48:35.0502717Z   File "/opt/miniconda/envs/mantidimaging_test/lib/python3.8/site-packages/pluggy/hooks.py", line 286 in __call__
2021-05-12T08:48:35.0503888Z   File "/opt/miniconda/envs/mantidimaging_test/lib/python3.8/site-packages/_pytest/main.py", line 323 in _main
2021-05-12T08:48:35.0505232Z   File "/opt/miniconda/envs/mantidimaging_test/lib/python3.8/site-packages/_pytest/main.py", line 269 in wrap_session
2021-05-12T08:48:35.0506428Z   File "/opt/miniconda/envs/mantidimaging_test/lib/python3.8/site-packages/_pytest/main.py", line 316 in pytest_cmdline_main
2021-05-12T08:48:35.0507640Z   File "/opt/miniconda/envs/mantidimaging_test/lib/python3.8/site-packages/pluggy/callers.py", line 187 in _multicall
2021-05-12T08:48:35.0508809Z   File "/opt/miniconda/envs/mantidimaging_test/lib/python3.8/site-packages/pluggy/manager.py", line 84 in <lambda>
2021-05-12T08:48:35.0510048Z   File "/opt/miniconda/envs/mantidimaging_test/lib/python3.8/site-packages/pluggy/manager.py", line 93 in _hookexec
2021-05-12T08:48:35.0511215Z   File "/opt/miniconda/envs/mantidimaging_test/lib/python3.8/site-packages/pluggy/hooks.py", line 286 in __call__
2021-05-12T08:48:35.0512352Z   File "/opt/miniconda/envs/mantidimaging_test/lib/python3.8/site-packages/xdist/remote.py", line 239 in <module>
2021-05-12T08:48:35.0513924Z   File "/opt/miniconda/envs/mantidimaging_test/lib/python3.8/site-packages/execnet/gateway_base.py", line 1084 in executetask
2021-05-12T08:48:35.0515135Z   File "/opt/miniconda/envs/mantidimaging_test/lib/python3.8/site-packages/execnet/gateway_base.py", line 220 in run
2021-05-12T08:48:35.0516457Z   File "/opt/miniconda/envs/mantidimaging_test/lib/python3.8/site-packages/execnet/gateway_base.py", line 285 in _perform_spawn
2021-05-12T08:48:35.0518055Z   File "/opt/miniconda/envs/mantidimaging_test/lib/python3.8/site-packages/execnet/gateway_base.py", line 267 in integrate_as_primary_thread
2021-05-12T08:48:35.0519738Z   File "/opt/miniconda/envs/mantidimaging_test/lib/python3.8/site-packages/execnet/gateway_base.py", line 1060 in serve
2021-05-12T08:48:35.0521122Z   File "/opt/miniconda/envs/mantidimaging_test/lib/python3.8/site-packages/execnet/gateway_base.py", line 1554 in serve
2021-05-12T08:48:35.0521944Z   File "<string>", line 8 in <module>
2021-05-12T08:48:35.0522378Z   File "<string>", line 1 in <module>
2021-05-12T08:48:35.6901265Z .[gw0] node down: Not properly terminated
2021-05-12T08:48:35.7377545Z F
2021-05-12T08:48:35.7378122Z replacing crashed worker gw0
2021-05-12T08:48:41.3034437Z ....................................
2021-05-12T08:48:41.3035688Z =================================== FAILURES ===================================
2021-05-12T08:48:41.3036689Z ____________ mantidimaging/core/operations/rebin/test/rebin_test.py ____________
2021-05-12T08:48:41.3042533Z [gw0] linux -- Python 3.8.8 /opt/miniconda/envs/mantidimaging_test/bin/python
2021-05-12T08:48:41.3044942Z worker 'gw0' crashed while running 'mantidimaging/core/operations/rebin/test/rebin_test.py::RebinTest::test_executed_xy_par_1024_1024'
2021-05-12T08:48:41.3045891Z =============================== warnings summary ===============================
2021-05-12T08:48:41.3047646Z ../miniconda/envs/mantidimaging_test/lib/python3.8/site-packages/skimage/util/dtype.py:226: 3 warnings
2021-05-12T08:48:41.3048984Z mantidimaging/core/operations/rotate_stack/test/rotate_stack_test.py: 20 warnings
2021-05-12T08:48:41.3049700Z mantidimaging/core/operations/rebin/test/rebin_test.py: 80 warnings
2021-05-12T08:48:41.3051396Z   /opt/miniconda/envs/mantidimaging_test/lib/python3.8/site-packages/skimage/util/dtype.py:226: DeprecationWarning: Converting `np.inexact` or `np.floating` to a dtype is deprecated. The current result is `float64` which is not strictly correct.
2021-05-12T08:48:41.3052511Z     dtypeobj_out = np.dtype(dtype)
2021-05-12T08:48:41.3052786Z 
2021-05-12T08:48:41.3054292Z ../miniconda/envs/mantidimaging_test/lib/python3.8/site-packages/tomopy/recon/acceleration.py:57
2021-05-12T08:48:41.3056001Z ../miniconda/envs/mantidimaging_test/lib/python3.8/site-packages/tomopy/recon/acceleration.py:57
2021-05-12T08:48:41.3057407Z ../miniconda/envs/mantidimaging_test/lib/python3.8/site-packages/tomopy/recon/acceleration.py:57
2021-05-12T08:48:41.3059632Z   /opt/miniconda/envs/mantidimaging_test/lib/python3.8/site-packages/tomopy/recon/acceleration.py:57: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
2021-05-12T08:48:41.3060646Z     import imp
2021-05-12T08:48:41.3060856Z 
2021-05-12T08:48:41.3061668Z mantidimaging/core/data/test/images_test.py::ImagesTest::test_copy_flip_axes
2021-05-12T08:48:41.3063494Z   /opt/miniconda/envs/mantidimaging_test/lib/python3.8/unittest/case.py:902: DeprecationWarning: elementwise comparison failed; this will raise an error in the future.
2021-05-12T08:48:41.3064865Z     if not first == second:
2021-05-12T08:48:41.3065124Z 
2021-05-12T08:48:41.3066056Z mantidimaging/core/data/test/images_test.py::ImagesTest::test_copy_flip_axes
2021-05-12T08:48:41.3067531Z   /opt/miniconda/envs/mantidimaging_test/lib/python3.8/unittest/case.py:918: DeprecationWarning: elementwise comparison failed; this will raise an error in the future.
2021-05-12T08:48:41.3068586Z     if not first != second:
2021-05-12T08:48:41.3068986Z 
2021-05-12T08:48:41.3069590Z mantidimaging/core/operations/monitor_normalisation/test/monitor_normalisation_test.py: 10 warnings
2021-05-12T08:48:41.3070740Z   /opt/mantidimaging/mantidimaging/core/operations/monitor_normalisation/monitor_normalisation.py:40: RuntimeWarning: divide by zero encountered in true_divide
2021-05-12T08:48:41.3071690Z     counts_val = counts.value / counts.value[0]
2021-05-12T08:48:41.3072010Z 
2021-05-12T08:48:41.3072599Z mantidimaging/core/operations/monitor_normalisation/test/monitor_normalisation_test.py: 10 warnings
2021-05-12T08:48:41.3073759Z   /opt/mantidimaging/mantidimaging/core/operations/monitor_normalisation/monitor_normalisation.py:40: RuntimeWarning: invalid value encountered in true_divide
2021-05-12T08:48:41.3074697Z     counts_val = counts.value / counts.value[0]
2021-05-12T08:48:41.3075016Z 
2021-05-12T08:48:41.3075503Z mantidimaging/core/operations/rescale/rescale_test.py: 10 warnings
2021-05-12T08:48:41.3076823Z   /opt/mantidimaging/mantidimaging/core/operations/rescale/rescale.py:40: RuntimeWarning: overflow encountered in float_scalars
2021-05-12T08:48:41.3078294Z     images.data *= (max_output / data_max)
2021-05-12T08:48:41.3078831Z 
2021-05-12T08:48:41.3079350Z mantidimaging/core/operations/rescale/rescale_test.py: 10 warnings
2021-05-12T08:48:41.3080532Z   /opt/mantidimaging/mantidimaging/core/operations/rescale/rescale.py:40: RuntimeWarning: invalid value encountered in multiply
2021-05-12T08:48:41.3081370Z     images.data *= (max_output / data_max)
2021-05-12T08:48:41.3082030Z 
2021-05-12T08:48:41.3082844Z -- Docs: https://docs.pytest.org/en/stable/warnings.html
2021-05-12T08:48:41.3083610Z =========================== short test summary info ============================
2021-05-12T08:48:41.3084282Z FAILED mantidimaging/core/operations/rebin/test/rebin_test.py::RebinTest::test_executed_xy_par_1024_1024
2021-05-12T08:48:41.3084987Z ========== 1 failed, 1220 passed, 10 skipped, 148 warnings in 24.44s ===========
2021-05-12T08:48:41.5262641Z ##[error]Process completed with exit code 1.
```

I think this is a better trace of the same issue
https://github.com/mantidproject/mantidimaging/pull/979/checks?check_run_id=2574369951

```
021-05-13T09:49:20.4460833Z =================================== FAILURES ===================================
2021-05-13T09:49:20.4461282Z ___________________ RebinTest.test_executed_xy_par_1024_1024 ___________________
2021-05-13T09:49:20.4462375Z [gw1] linux -- Python 3.8.10 /opt/miniconda/envs/mantidimaging_test/bin/python
2021-05-13T09:49:20.4462756Z 
2021-05-13T09:49:20.4463609Z self = <mantidimaging.core.operations.rebin.test.rebin_test.RebinTest testMethod=test_executed_xy_par_1024_1024>
2021-05-13T09:49:20.4464402Z 
2021-05-13T09:49:20.4464731Z     def test_executed_xy_par_1024_1024(self):
2021-05-13T09:49:20.4465217Z >       self.do_execute_xy(True, (1024, 1024))
2021-05-13T09:49:20.4465452Z 
2021-05-13T09:49:20.4465834Z mantidimaging/core/operations/rebin/test/rebin_test.py:80: 
2021-05-13T09:49:20.4466275Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2021-05-13T09:49:20.4466940Z mantidimaging/core/operations/rebin/test/rebin_test.py:101: in do_execute_xy
2021-05-13T09:49:20.4467579Z     result = RebinFilter.filter_func(images, rebin_param=val, mode=mode)
2021-05-13T09:49:20.4468197Z mantidimaging/core/operations/rebin/rebin.py:53: in filter_func
2021-05-13T09:49:20.4468764Z     empty_resized_data = _create_reshaped_array(images, rebin_param)
2021-05-13T09:49:20.4469573Z mantidimaging/core/operations/rebin/rebin.py:167: in _create_reshaped_array
2021-05-13T09:49:20.4470166Z     return pu.create_array(shape, images.dtype)
2021-05-13T09:49:20.4470719Z mantidimaging/core/parallel/utility.py:45: in create_array
2021-05-13T09:49:20.4471249Z     return _create_shared_array(shape, dtype)
2021-05-13T09:49:20.4471791Z mantidimaging/core/parallel/utility.py:77: in _create_shared_array
2021-05-13T09:49:20.4472296Z     array = Array(ctype, length)
2021-05-13T09:49:20.4472884Z ../miniconda/envs/mantidimaging_test/lib/python3.8/multiprocessing/context.py:141: in Array
2021-05-13T09:49:20.4473592Z     return Array(typecode_or_type, size_or_initializer, lock=lock,
2021-05-13T09:49:20.4474463Z ../miniconda/envs/mantidimaging_test/lib/python3.8/multiprocessing/sharedctypes.py:93: in Array
2021-05-13T09:49:20.4475068Z     lock = ctx.RLock()
2021-05-13T09:49:20.4475628Z ../miniconda/envs/mantidimaging_test/lib/python3.8/multiprocessing/context.py:73: in RLock
2021-05-13T09:49:20.4476244Z     return RLock(ctx=self.get_context())
2021-05-13T09:49:20.4476890Z ../miniconda/envs/mantidimaging_test/lib/python3.8/multiprocessing/synchronize.py:187: in __init__
2021-05-13T09:49:20.4477558Z     SemLock.__init__(self, RECURSIVE_MUTEX, 1, 1, ctx=ctx)
2021-05-13T09:49:20.4477972Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2021-05-13T09:49:20.4478183Z 
2021-05-13T09:49:20.4478546Z self = <RLock(unknown, unknown)>, kind = 0, value = 1, maxvalue = 1
2021-05-13T09:49:20.4478856Z 
2021-05-13T09:49:20.4479193Z     def __init__(self, kind, value, maxvalue, *, ctx):
2021-05-13T09:49:20.4479671Z         if ctx is None:
2021-05-13T09:49:20.4480085Z             ctx = context._default_context.get_context()
2021-05-13T09:49:20.4480517Z         name = ctx.get_start_method()
2021-05-13T09:49:20.4481152Z         unlink_now = sys.platform == 'win32' or name == 'fork'
2021-05-13T09:49:20.4481538Z         for i in range(100):
2021-05-13T09:49:20.4481824Z             try:
2021-05-13T09:49:20.4482262Z >               sl = self._semlock = _multiprocessing.SemLock(
2021-05-13T09:49:20.4482812Z                     kind, value, maxvalue, self._make_name(),
2021-05-13T09:49:20.4483189Z                     unlink_now)
2021-05-13T09:49:20.4483581Z E                   OSError: [Errno 28] No space left on device
2021-05-13T09:49:20.4483858Z 
2021-05-13T09:49:20.4484402Z ../miniconda/envs/mantidimaging_test/lib/python3.8/multiprocessing/synchronize.py:57: OSError
2021-05-13T09:49:20.4485282Z ----------------------------- Captured stdout call -----------------------------
2021-05-13T09:49:20.4486463Z 2021-05-13 09:49:14,695 [mantidimaging.core.parallel.utility:L75] INFO: Requested shared array with shape=(15, 8, 10), length=1200, size=4800, dtype=<class 'numpy.float32'>
2021-05-13T09:49:20.4488023Z 2021-05-13 09:49:14,696 [mantidimaging.core.parallel.utility:L75] INFO: Requested shared array with shape=(15, 1024, 1024), length=15728640, size=62914560, dtype=<class 'numpy.float32'>
2021-05-13T09:49:20.4489115Z ------------------------------ Captured log call -------------------------------
2021-05-13T09:49:20.4490273Z INFO     mantidimaging.core.parallel.utility:utility.py:75 Requested shared array with shape=(15, 8, 10), length=1200, size=4800, dtype=<class 'numpy.float32'>
2021-05-13T09:49:20.4491909Z INFO     mantidimaging.core.parallel.utility:utility.py:75 Requested shared array with shape=(15, 1024, 1024), length=15728640, size=62914560, dtype=<class 'numpy.float32'>
2021-05-13T09:49:20.4492824Z __________________ RebinTest.test_not_executed_rebin_negative __________________
2021-05-13T09:49:20.4493530Z [gw1] linux -- Python 3.8.10 /opt/miniconda/envs/mantidimaging_test/bin/python
2021-05-13T09:49:20.4493952Z 
2021-05-13T09:49:20.4494809Z self = <mantidimaging.core.operations.rebin.test.rebin_test.RebinTest testMethod=test_not_executed_rebin_negative>
2021-05-13T09:49:20.4495576Z 
2021-05-13T09:49:20.4495911Z     def test_not_executed_rebin_negative(self):
2021-05-13T09:49:20.4496503Z >       images = th.generate_images()
2021-05-13T09:49:20.4496757Z 
2021-05-13T09:49:20.4497169Z mantidimaging/core/operations/rebin/test/rebin_test.py:24: 
2021-05-13T09:49:20.4497651Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2021-05-13T09:49:20.4498152Z mantidimaging/test_helpers/unit_test_helper.py:39: in generate_images
2021-05-13T09:49:20.4498689Z     d = pu.create_array(shape, dtype)
2021-05-13T09:49:20.4499185Z mantidimaging/core/parallel/utility.py:45: in create_array
2021-05-13T09:49:20.4499713Z     return _create_shared_array(shape, dtype)
2021-05-13T09:49:20.4500253Z mantidimaging/core/parallel/utility.py:77: in _create_shared_array
2021-05-13T09:49:20.4500929Z     array = Array(ctype, length)
2021-05-13T09:49:20.4501534Z ../miniconda/envs/mantidimaging_test/lib/python3.8/multiprocessing/context.py:141: in Array
2021-05-13T09:49:20.4502255Z     return Array(typecode_or_type, size_or_initializer, lock=lock,
2021-05-13T09:49:20.4502999Z ../miniconda/envs/mantidimaging_test/lib/python3.8/multiprocessing/sharedctypes.py:93: in Array
2021-05-13T09:49:20.4503612Z     lock = ctx.RLock()
2021-05-13T09:49:20.4504196Z ../miniconda/envs/mantidimaging_test/lib/python3.8/multiprocessing/context.py:73: in RLock
2021-05-13T09:49:20.4504820Z     return RLock(ctx=self.get_context())
2021-05-13T09:49:20.4505486Z ../miniconda/envs/mantidimaging_test/lib/python3.8/multiprocessing/synchronize.py:187: in __init__
2021-05-13T09:49:20.4506170Z     SemLock.__init__(self, RECURSIVE_MUTEX, 1, 1, ctx=ctx)
2021-05-13T09:49:20.4506594Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2021-05-13T09:49:20.4506870Z 
2021-05-13T09:49:20.4507248Z self = <RLock(unknown, unknown)>, kind = 0, value = 1, maxvalue = 1
2021-05-13T09:49:20.4507567Z 
2021-05-13T09:49:20.4507925Z     def __init__(self, kind, value, maxvalue, *, ctx):
2021-05-13T09:49:20.4508304Z         if ctx is None:
2021-05-13T09:49:20.4508722Z             ctx = context._default_context.get_context()
2021-05-13T09:49:20.4509168Z         name = ctx.get_start_method()
2021-05-13T09:49:20.4509783Z         unlink_now = sys.platform == 'win32' or name == 'fork'
2021-05-13T09:49:20.4510180Z         for i in range(100):
2021-05-13T09:49:20.4510472Z             try:
2021-05-13T09:49:20.4510919Z >               sl = self._semlock = _multiprocessing.SemLock(
2021-05-13T09:49:20.4511480Z                     kind, value, maxvalue, self._make_name(),
2021-05-13T09:49:20.4511865Z                     unlink_now)
2021-05-13T09:49:20.4512263Z E                   OSError: [Errno 28] No space left on device
2021-05-13T09:49:20.4512550Z 
2021-05-13T09:49:20.4513110Z ../miniconda/envs/mantidimaging_test/lib/python3.8/multiprocessing/synchronize.py:57: OSError
2021-05-13T09:49:20.4514005Z ----------------------------- Captured stdout call -----------------------------
2021-05-13T09:49:20.4525444Z 2021-05-13 09:49:14,789 [mantidimaging.core.parallel.utility:L75] INFO: Requested shared array with shape=(10, 8, 10), length=800, size=3200, dtype=<class 'numpy.float32'>
2021-05-13T09:49:20.4526568Z ------------------------------ Captured log call -------------------------------
2021-05-13T09:49:20.4527774Z INFO     mantidimaging.core.parallel.utility:utility.py:75 Requested shared array with shape=(10, 8, 10), length=800, size=3200, dtype=<class 'numpy.float32'>
2021-05-13T09:49:20.4528784Z ______________ RebinTest.test_execute_wrapper_return_is_runnable _______________
2021-05-13T09:49:20.4529579Z [gw1] linux -- Python 3.8.10 /opt/miniconda/envs/mantidimaging_test/bin/python
2021-05-13T09:49:20.4529967Z 
2021-05-13T09:49:20.4531004Z self = <mantidimaging.core.operations.rebin.test.rebin_test.RebinTest testMethod=test_execute_wrapper_return_is_runnable>
2021-05-13T09:49:20.4531873Z 
2021-05-13T09:49:20.4532263Z     def test_execute_wrapper_return_is_runnable(self):
2021-05-13T09:49:20.4532643Z         """
2021-05-13T09:49:20.4533176Z         Test that the partial returned by execute_wrapper can be executed (kwargs are named correctly)
2021-05-13T09:49:20.4533685Z         """
2021-05-13T09:49:20.4534044Z         rebin_to_dimensions_radio = mock.Mock()
2021-05-13T09:49:20.4534609Z         rebin_to_dimensions_radio.isChecked = mock.Mock(return_value=False)
2021-05-13T09:49:20.4535164Z         rebin_by_factor_radio = mock.Mock()
2021-05-13T09:49:20.4535695Z         rebin_by_factor_radio.isChecked = mock.Mock(return_value=True)
2021-05-13T09:49:20.4536172Z         factor = mock.Mock()
2021-05-13T09:49:20.4536714Z         factor.value = mock.Mock(return_value=0)
2021-05-13T09:49:20.4537203Z         mode_field = mock.Mock()
2021-05-13T09:49:20.4537652Z         mode_field.currentText = mock.Mock(return_value=0)
2021-05-13T09:49:20.4538338Z         execute_func = RebinFilter.execute_wrapper(rebin_to_dimensions_radio=rebin_to_dimensions_radio,
2021-05-13T09:49:20.4538992Z                                                    rebin_by_factor_radio=rebin_by_factor_radio,
2021-05-13T09:49:20.4539394Z                                                    factor=factor,
2021-05-13T09:49:20.4539762Z                                                    mode_field=mode_field)
2021-05-13T09:49:20.4540063Z     
2021-05-13T09:49:20.4540358Z >       images = th.generate_images()
2021-05-13T09:49:20.4540597Z 
2021-05-13T09:49:20.4540983Z mantidimaging/core/operations/rebin/test/rebin_test.py:136: 
2021-05-13T09:49:20.4541422Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2021-05-13T09:49:20.4541895Z mantidimaging/test_helpers/unit_test_helper.py:39: in generate_images
2021-05-13T09:49:20.4542392Z     d = pu.create_array(shape, dtype)
2021-05-13T09:49:20.4542955Z mantidimaging/core/parallel/utility.py:45: in create_array
2021-05-13T09:49:20.4543449Z     return _create_shared_array(shape, dtype)
2021-05-13T09:49:20.4543975Z mantidimaging/core/parallel/utility.py:77: in _create_shared_array
2021-05-13T09:49:20.4544443Z     array = Array(ctype, length)
2021-05-13T09:49:20.4545016Z ../miniconda/envs/mantidimaging_test/lib/python3.8/multiprocessing/context.py:141: in Array
2021-05-13T09:49:20.4545670Z     return Array(typecode_or_type, size_or_initializer, lock=lock,
2021-05-13T09:49:20.4546370Z ../miniconda/envs/mantidimaging_test/lib/python3.8/multiprocessing/sharedctypes.py:93: in Array
2021-05-13T09:49:20.4546943Z     lock = ctx.RLock()
2021-05-13T09:49:20.4547469Z ../miniconda/envs/mantidimaging_test/lib/python3.8/multiprocessing/context.py:73: in RLock
2021-05-13T09:49:20.4548050Z     return RLock(ctx=self.get_context())
2021-05-13T09:49:20.4548748Z ../miniconda/envs/mantidimaging_test/lib/python3.8/multiprocessing/synchronize.py:187: in __init__
2021-05-13T09:49:20.4549585Z     SemLock.__init__(self, RECURSIVE_MUTEX, 1, 1, ctx=ctx)
2021-05-13T09:49:20.4550006Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2021-05-13T09:49:20.4550205Z 
2021-05-13T09:49:20.4550582Z self = <RLock(unknown, unknown)>, kind = 0, value = 1, maxvalue = 1
2021-05-13T09:49:20.4550890Z 
2021-05-13T09:49:20.4551226Z     def __init__(self, kind, value, maxvalue, *, ctx):
2021-05-13T09:49:20.4551607Z         if ctx is None:
2021-05-13T09:49:20.4552004Z             ctx = context._default_context.get_context()
2021-05-13T09:49:20.4552447Z         name = ctx.get_start_method()
2021-05-13T09:49:20.4553054Z         unlink_now = sys.platform == 'win32' or name == 'fork'
2021-05-13T09:49:20.4553436Z         for i in range(100):
2021-05-13T09:49:20.4553720Z             try:
2021-05-13T09:49:20.4554154Z >               sl = self._semlock = _multiprocessing.SemLock(
2021-05-13T09:49:20.4557251Z                     kind, value, maxvalue, self._make_name(),
2021-05-13T09:49:20.4557634Z                     unlink_now)
2021-05-13T09:49:20.4558115Z E                   OSError: [Errno 28] No space left on device
2021-05-13T09:49:20.4558400Z 
2021-05-13T09:49:20.4558944Z ../miniconda/envs/mantidimaging_test/lib/python3.8/multiprocessing/synchronize.py:57: OSError
2021-05-13T09:49:20.4559873Z ----------------------------- Captured stdout call -----------------------------
2021-05-13T09:49:20.4561044Z 2021-05-13 09:49:14,814 [mantidimaging.core.parallel.utility:L75] INFO: Requested shared array with shape=(10, 8, 10), length=800, size=3200, dtype=<class 'numpy.float32'>
2021-05-13T09:49:20.4562132Z ------------------------------ Captured log call -------------------------------
2021-05-13T09:49:20.4563302Z INFO     mantidimaging.core.parallel.utility:utility.py:75 Requested shared array with shape=(10, 8, 10), length=800, size=3200, dtype=<class 'numpy.float32'>
2021-05-13T09:49:20.4564414Z ____________________ RebinTest.test_execute_argument_order _____________________
2021-05-13T09:49:20.4565188Z [gw1] linux -- Python 3.8.10 /opt/miniconda/envs/mantidimaging_test/bin/python
2021-05-13T09:49:20.4565573Z 
2021-05-13T09:49:20.4566465Z self = <mantidimaging.core.operations.rebin.test.rebin_test.RebinTest testMethod=test_execute_argument_order>
2021-05-13T09:49:20.4567594Z ps_mock = <MagicMock name='ps' id='140157165859312'>
2021-05-13T09:49:20.4567860Z 
2021-05-13T09:49:20.4568552Z     @mock.patch("mantidimaging.core.operations.rebin.rebin.ps")
2021-05-13T09:49:20.4569389Z     def test_execute_argument_order(self, ps_mock):
2021-05-13T09:49:20.4569752Z     
2021-05-13T09:49:20.4570088Z >       images = th.generate_images()
2021-05-13T09:49:20.4570338Z 
2021-05-13T09:49:20.4570769Z mantidimaging/core/operations/rebin/test/rebin_test.py:147: 
2021-05-13T09:49:20.4571244Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2021-05-13T09:49:20.4571744Z mantidimaging/test_helpers/unit_test_helper.py:39: in generate_images
2021-05-13T09:49:20.4572373Z     d = pu.create_array(shape, dtype)
2021-05-13T09:49:20.4572886Z mantidimaging/core/parallel/utility.py:45: in create_array
2021-05-13T09:49:20.4573430Z     return _create_shared_array(shape, dtype)
2021-05-13T09:49:20.4573985Z mantidimaging/core/parallel/utility.py:77: in _create_shared_array
2021-05-13T09:49:20.4574504Z     array = Array(ctype, length)
2021-05-13T09:49:20.4575107Z ../miniconda/envs/mantidimaging_test/lib/python3.8/multiprocessing/context.py:141: in Array
2021-05-13T09:49:20.4575830Z     return Array(typecode_or_type, size_or_initializer, lock=lock,
2021-05-13T09:49:20.4576584Z ../miniconda/envs/mantidimaging_test/lib/python3.8/multiprocessing/sharedctypes.py:93: in Array
2021-05-13T09:49:20.4577190Z     lock = ctx.RLock()
2021-05-13T09:49:20.4577779Z ../miniconda/envs/mantidimaging_test/lib/python3.8/multiprocessing/context.py:73: in RLock
2021-05-13T09:49:20.4578400Z     return RLock(ctx=self.get_context())
2021-05-13T09:49:20.4579071Z ../miniconda/envs/mantidimaging_test/lib/python3.8/multiprocessing/synchronize.py:187: in __init__
2021-05-13T09:49:20.4579765Z     SemLock.__init__(self, RECURSIVE_MUTEX, 1, 1, ctx=ctx)
2021-05-13T09:49:20.4580190Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2021-05-13T09:49:20.4580407Z 
2021-05-13T09:49:20.4580780Z self = <RLock(unknown, unknown)>, kind = 0, value = 1, maxvalue = 1
2021-05-13T09:49:20.4581096Z 
2021-05-13T09:49:20.4581453Z     def __init__(self, kind, value, maxvalue, *, ctx):
2021-05-13T09:49:20.4581835Z         if ctx is None:
2021-05-13T09:49:20.4582254Z             ctx = context._default_context.get_context()
2021-05-13T09:49:20.4582697Z         name = ctx.get_start_method()
2021-05-13T09:49:20.4583296Z         unlink_now = sys.platform == 'win32' or name == 'fork'
2021-05-13T09:49:20.4583692Z         for i in range(100):
2021-05-13T09:49:20.4583981Z             try:
2021-05-13T09:49:20.4584546Z >               sl = self._semlock = _multiprocessing.SemLock(
2021-05-13T09:49:20.4585104Z                     kind, value, maxvalue, self._make_name(),
2021-05-13T09:49:20.4585548Z                     unlink_now)
2021-05-13T09:49:20.4586031Z E                   OSError: [Errno 28] No space left on device
2021-05-13T09:49:20.4586297Z 
2021-05-13T09:49:20.4586811Z ../miniconda/envs/mantidimaging_test/lib/python3.8/multiprocessing/synchronize.py:57: OSError
2021-05-13T09:49:20.4587649Z ----------------------------- Captured stdout call -----------------------------
2021-05-13T09:49:20.4588769Z 2021-05-13 09:49:14,839 [mantidimaging.core.parallel.utility:L75] INFO: Requested shared array with shape=(10, 8, 10), length=800, size=3200, dtype=<class 'numpy.float32'>
2021-05-13T09:49:20.4589793Z ------------------------------ Captured log call -------------------------------
2021-05-13T09:49:20.4590905Z INFO     mantidimaging.core.parallel.utility:utility.py:75 Requested shared array with shape=(10, 8, 10), length=800, size=3200, dtype=<class 'numpy.float32'>
2021-05-13T09:49:20.4591724Z =============================== warnings summary ===============================
2021-05-13T09:49:20.4592489Z ../miniconda/envs/mantidimaging_test/lib/python3.8/site-packages/skimage/util/dtype.py:226: 2 warnings
2021-05-13T09:49:20.4593236Z mantidimaging/core/operations/rotate_stack/test/rotate_stack_test.py: 20 warnings
2021-05-13T09:49:20.4593870Z mantidimaging/core/operations/rebin/test/rebin_test.py: 80 warnings
2021-05-13T09:49:20.4595610Z   /opt/miniconda/envs/mantidimaging_test/lib/python3.8/site-packages/skimage/util/dtype.py:226: DeprecationWarning: Converting `np.inexact` or `np.floating` to a dtype is deprecated. The current result is `float64` which is not strictly correct.
2021-05-13T09:49:20.4596668Z     dtypeobj_out = np.dtype(dtype)
2021-05-13T09:49:20.4596913Z 
2021-05-13T09:49:20.4597660Z ../miniconda/envs/mantidimaging_test/lib/python3.8/site-packages/tomopy/recon/acceleration.py:57
2021-05-13T09:49:20.4598693Z ../miniconda/envs/mantidimaging_test/lib/python3.8/site-packages/tomopy/recon/acceleration.py:57
2021-05-13T09:49:20.4600304Z   /opt/miniconda/envs/mantidimaging_test/lib/python3.8/site-packages/tomopy/recon/acceleration.py:57: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
2021-05-13T09:49:20.4601289Z     import imp
2021-05-13T09:49:20.4601486Z 
2021-05-13T09:49:20.4601920Z mantidimaging/core/operations/rescale/rescale_test.py: 10 warnings
2021-05-13T09:49:20.4602793Z   /opt/mantidimaging/mantidimaging/core/operations/rescale/rescale.py:40: RuntimeWarning: overflow encountered in float_scalars
2021-05-13T09:49:20.4603531Z     images.data *= (max_output / data_max)
2021-05-13T09:49:20.4603792Z 
2021-05-13T09:49:20.4604237Z mantidimaging/core/operations/rescale/rescale_test.py: 10 warnings
2021-05-13T09:49:20.4605096Z   /opt/mantidimaging/mantidimaging/core/operations/rescale/rescale.py:40: RuntimeWarning: invalid value encountered in multiply
2021-05-13T09:49:20.4605847Z     images.data *= (max_output / data_max)
2021-05-13T09:49:20.4606111Z 
2021-05-13T09:49:20.4607270Z mantidimaging/core/data/test/images_test.py::ImagesTest::test_copy_flip_axes
2021-05-13T09:49:20.4608348Z   /opt/miniconda/envs/mantidimaging_test/lib/python3.8/unittest/case.py:902: DeprecationWarning: elementwise comparison failed; this will raise an error in the future.
2021-05-13T09:49:20.4609091Z     if not first == second:
2021-05-13T09:49:20.4609298Z 
2021-05-13T09:49:20.4609811Z mantidimaging/core/data/test/images_test.py::ImagesTest::test_copy_flip_axes
2021-05-13T09:49:20.4610729Z   /opt/miniconda/envs/mantidimaging_test/lib/python3.8/unittest/case.py:918: DeprecationWarning: elementwise comparison failed; this will raise an error in the future.
2021-05-13T09:49:20.4611480Z     if not first != second:
2021-05-13T09:49:20.4611686Z 
2021-05-13T09:49:20.4612204Z mantidimaging/core/operations/monitor_normalisation/test/monitor_normalisation_test.py: 10 warnings
2021-05-13T09:49:20.4613237Z   /opt/mantidimaging/mantidimaging/core/operations/monitor_normalisation/monitor_normalisation.py:40: RuntimeWarning: divide by zero encountered in true_divide
2021-05-13T09:49:20.4614163Z     counts_val = counts.value / counts.value[0]
2021-05-13T09:49:20.4614435Z 
2021-05-13T09:49:20.4614967Z mantidimaging/core/operations/monitor_normalisation/test/monitor_normalisation_test.py: 10 warnings
2021-05-13T09:49:20.4616005Z   /opt/mantidimaging/mantidimaging/core/operations/monitor_normalisation/monitor_normalisation.py:40: RuntimeWarning: invalid value encountered in true_divide
2021-05-13T09:49:20.4617028Z     counts_val = counts.value / counts.value[0]
2021-05-13T09:49:20.4617486Z 
2021-05-13T09:49:20.4618157Z -- Docs: https://docs.pytest.org/en/stable/warnings.html
2021-05-13T09:49:20.4618743Z =========================== short test summary info ============================
2021-05-13T09:49:20.4619395Z FAILED mantidimaging/core/operations/rebin/test/rebin_test.py::RebinTest::test_executed_xy_par_1024_1024
2021-05-13T09:49:20.4620266Z FAILED mantidimaging/core/operations/rebin/test/rebin_test.py::RebinTest::test_not_executed_rebin_negative
2021-05-13T09:49:20.4621208Z FAILED mantidimaging/core/operations/rebin/test/rebin_test.py::RebinTest::test_execute_wrapper_return_is_runnable
2021-05-13T09:49:20.4622117Z FAILED mantidimaging/core/operations/rebin/test/rebin_test.py::RebinTest::test_execute_argument_order
2021-05-13T09:49:20.4622822Z ========== 4 failed, 1217 passed, 10 skipped, 146 warnings in 16.71s ===========
```

--- a/mantidimaging/core/operations/rebin/test/rebin_test.py
+++ b/mantidimaging/core/operations/rebin/test/rebin_test.py
@@ -76,17 +76,11 @@ class RebinTest(unittest.TestCase):
     def test_executed_xy_par_512_256(self):
         self.do_execute_xy(True, (512, 256))
 
-    def test_executed_xy_par_1024_1024(self):
-        self.do_execute_xy(True, (1024, 1024))
-
     def test_executed_xy_seq_128_256(self):
         self.do_execute_xy(False, (128, 256))
 
     def test_executed_xy_seq_512_256(self):
         self.do_execute_xy(False, (512, 256))
-
-    def test_executed_xy_seq_1024_1024(self):
-        self.do_execute_xy(False, (1024, 1024))
 
     def do_execute_xy(self, is_parallel: bool, val=(512, 512)):
         if is_parallel:

